pkg/scheduler/schedule_one.go

```diff
@@ -94,7 +94,9 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 
 	scheduleResult, assumedPodInfo, err := sched.schedulingCycle(schedulingCycleCtx, state, fwk, podInfo, start, podsToActivate)
```

Comment:
Can we return status directly, then we can remove the err. For status will contain everything about the error message. For binding cycle, it's really easy, but for scheduling cycle, we may have to construct status to return. 

I haven't try it yet, but I think it will work.

Reply:
Do you think we should also return status instead of error in function `SchedulePod`?

Reply:
Good question, I think it worths a shot, but maybe we can leave it a followup.